backend/pkg/helm/release_test.go

```diff
@@ -35,17 +40,57 @@ func GetClient(t *testing.T, clusterName string) clientcmd.ClientConfig {
 	return clientcmd.NewInteractiveClientConfig(*config, clusterName, nil, nil, nil)
 }
 
-func TestInstallRelease(t *testing.T) {
-	k8sclient := GetClient(t, "minikube")
+func getStatus(t *testing.T, ch cache.Cache, action string, releaseName string) (string, error) {
+	t.Helper()
 
-	helmHandler, err := NewHandlerWithSettings(k8sclient, "default", settings)
+	statusVal, err := ch.Get(context.Background(), "helm_"+action+"_"+releaseName)
 	require.NoError(t, err)
 
-	// add headlmap repo
-	err = AddRepository("headlamp", "https://headlamp-k8s.github.io/headlamp/", settings)
+	valueBytes, err := json.Marshal(statusVal)
+	require.NoError(t, err)
+
+	var status struct {
+		Status string
+		Err    error
+	}
+
+	err = json.Unmarshal(valueBytes, &status)
 	require.NoError(t, err)
 
-	// TODO: uninstall release if it already exists
+	return status.Status, status.Err
+}
+
+//nolint:unparam
+func pingStatusTillSuccess(t *testing.T, action, releaseName string, cache cache.Cache) {
+	t.Helper()
+
+	now := time.Now()
+
+	for {
+		if time.Since(now) > time.Minute {
+			t.Fatal("timed out waiting for status to be success")
+		}
+
+		status, err := getStatus(t, cache, action, releaseName)
+		require.NoError(t, err)
+
+		if status == "success" {
+			break
+		}
+
```

Comment:
Probably a sleep here, or it might busyloop for 1 minute when it's failing or slow.